universe/components.py

```diff
@@ -0,0 +1,41 @@
+from . import fields
+
+
+class MetaComponent(type):
+    def __new__(cls, name, bases, attrs, **kwargs):
+        super_new = super().__new__
+
+        new_attrs = {'_fields': {}}
+        for name, f in attrs.items():
+            if isinstance(f, fields.Field):
+                new_attrs['_fields'][name] = f
+            else:
+                new_attrs[name] = f
+
+        new_class = super_new(cls, name, bases, new_attrs, **kwargs)
+        return new_class
+
+
+class Component(metaclass=MetaComponent):
+    def serialize(self, data):
+        return {name: data[name] for name, field in self._fields.items()
```

Comment:
Thoughts: we should probably throw a customized exception type if there is a field not marked `required=False` that fails to have the value in the data dict.